cranelift/codegen/src/isa/zkasm/lower.isle

```diff
@@ -31,8 +31,8 @@
 (rule 0 (lower (has_type (ty_int_ref_scalar_64 ty) (iadd x y)))
   (rv_add x y))
 
-(rule 1 (lower (iadd (imm32_from_value x) (imm32_from_value y)))
-  (zk_add x y))
+;; (rule 1 (lower (iadd (imm32_from_value x) (imm32_from_value y)))
+;;   (zk_add x y))
```

Comment:
Idk why but this rule was matching
```
i64.const 1
i64.const 1
i64.add
```
Disabled for now, it also increased amount of passing spectests.

Reply:
Why shouldn’t it have matched? `1` fits into a `32-bit` immediate (which qualifies this for `1 + 1 => R` form, and the result is going to be `2` either way. Hmm… though I guess that will be problematic if this matches for `0xFFFF_FFFF + 2` which… given the unclear limb computation rules does not work. This should be matching for 31-bit immediates…

Reply:
It's a good catch. We should aim to restore this logic for 31-bit immediate as Simonas points out, as this is an important optimization - let's file an issue for this and add a TODO here.